experimental/FTheoryTools/src/HypersurfaceModels/constructors.jl

```diff
@@ -282,7 +295,49 @@ function hypersurface_model(auxiliary_base_vars::Vector{String}, auxiliary_base_
   model = HypersurfaceModel(toric_covered_scheme(auxiliary_base_space), toric_covered_scheme(auxiliary_ambient_space), toric_covered_scheme(fiber_ambient_space), hypersurface_equation)
   set_attribute!(model, :base_fully_specified, false)
   return model
+end
+
+
+function _construct_generic_sample(base_vars::Vector{String}, base_grading::Matrix{Int64}, d::Int, fiber_ambient_space::NormalToricVariety, D1::Vector{Int64}, D2::Vector{Int64}, p::MPolyRingElem)
+  # Convert Weierstrass sections into polynomials of the auxiliary base
+  base_space = family_of_spaces(PolynomialRing(QQ, base_vars, cached = false)[1], base_grading, d)
   
+  # Construct ambient space
+  ambient_space_vars = vcat(base_vars, coordinate_names(fiber_ambient_space))
+  coordinate_ring_ambient_space = PolynomialRing(QQ, ambient_space_vars, cached = false)[1]
+  w = vcat([k.coeff for k in cox_ring(fiber_ambient_space_space).d])
+  ambient_space_grading = zero_matrix(Int, nrows(base_grading)+nrows(S),ncols(base_grading)+ncols(w))
+  for i in 1:nrows(base_grading)
+    for j in 1:ncols(base_grading)
+      ambient_space_grading[i,j] = base_grading[i,j]
+    end
+  end
+  for i in 1:nrows(w)
+    for j in 1:ncols(w)
+      ambient_space_grading[nrows(base_grading)+i,ncols(base_grading)+j] = w[i,j]
+    end
+  end
+  for i in 1:rows(w)
+    ambient_space_grading[i,ncols(base_grading)+1] = D1[i]
+    ambient_space_grading[i,ncols(base_grading)+2] = D2[i]
+  end
```

Comment:
Should this not be
```
for i in 1:ncols(base_grading)
    ambient_space_grading[nrows(base_grading)+1, i] = D1[i]
    ambient_space_grading[nrows(base_grading)+2, i] = D2[i]
  end
```

Before going through this loop, we have a matrix with a block in the top left that is the base_grading, and a block in the bottom right that is the ambient fiber space grading. What I think we want to do with this loop is modify the lower left block to include the twisting of the fiber ambient space over the base. We only need to specify the twisting of two of the fiber coordinates, which is what D1 and D2 are for, and they should have the same length as the number of rows of the base_grading. Please correct me if I'm wrong.

Regardless, if I try to reconstruct the above three loops for the Krause, Mayrhofer, Weigand model, this last loop gives me an out-of-bounds error, so something here seems to be wrong.

Reply:
Assuming my above comment is correct, I think the result of the above loops could instead be achieved as follows, starting from the definition of `w`:
```
w = Matrix{Int64}(vcat([k.coeff for k in cox_ring(fiber_ambient_space).d]))
z_block = zeros(Int64, nrows(base_grading), ncols(w))
D_block = [D1'; D2'; zeros(Int64, 1, ncols(base_grading))]

ambient_space_grading = hvcat(2, base_grading, z_block, D_block, w)
```
I leave it up to you if this is better.

Reply:
If you want to be even more terse, the last line is also achieved with
```ambient_space_grading = [base_grading z_block; D_block w)```